### DISCUSSION TITLE
Suggest your valuable ideas here.

### DISCUSSION THREAD

**Discussion by HARSH-SHETH** [HARSH-SHETH/miku#2] [Ideas]
My Ideas: 
1. Implement [waifu](https://waifu.pics/docs) (SFW only) api.
2. Implement reverse image search feature.
3. Implement sticker generation from any image, gif, video and youtube-url.

**Comment by HARSH-SHETH** [comment]
4. Implement feature to reveal last deleted message in a group or private conversation.
5. Implement [random fuck off](https://foaas.com/) api.

**Comment by SarthakGupta3** [comment]
6. Implement feature to show "important" message in the group.
7. Implement feature to delete multiple message from the chat of that user based on count user provides (eg - last 5 messages).
8. Polling Feature.

**Reply by HARSH-SHETH** [reply]
> Implement feature to show "important" message in the group.

!minna command is created for this purpose only(To tag everyone whenever there is a important message).
> Implement feature to delete multiple message from the chat of that user based on count user provides (eg - last 5 messages).

This feature can be implemented :+1: .
> Polling Feature.

Cool. Send a pull request once u are ready.

**Comment by moulikbhardwaj** [comment]
### Support for class schedule
**Idea:**
- implement feature to show today's class schedule, or tomorrow's class schedule
- Give details about current and next class (subject and link)
- Implement functionality to temporarily modify the schedule (class shifted or cancelled)

**Possible Implementation**
`!miku schedule` will respond with today's schedule, or tomorrow's if no more classes are there today
`!miku class` will respond with current and next (if exist) class's subject name, subject code, time and meet link.
`!miku CR SUBJECT-CODE cancel [date]` will cancel that class for today if date is empty, or cancel class on given date
`!miku CR SUBJECT-CODE shift [from-date] [to-date]` will shift that class for today if date is empty, or on given date, to [to-date]
`!miku CR SUBJECT-CODE1 SUBJECT-CODE2 swap [date]` will swap classes for today if date is empty, or on given date

The sub-commands following CR must require authorization of admin or CR/GR.

**Reply by HARSH-SHETH** [reply]
!miku class is implemented.
